contracts/modules/RelayerModule.sol

```diff
@@ -0,0 +1,356 @@
+// Copyright (C) 2018  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.10;
+pragma experimental ABIEncoderV2;
+
+import "./common/Utils.sol";
+import "./common/BaseModule.sol";
+import "./common/GuardianUtils.sol";
+import "./common/LimitUtils.sol";
+import "../infrastructure/storage/ILimitStorage.sol";
+import "../infrastructure/storage/ITokenPriceStorage.sol";
+
+/**
+ * @title RelayerModule
+ * @dev Module to execute transactions signed by eth-less accounts and sent by a relayer.
+ * @author Julien Niset <julien@argent.xyz>, Olivier VDB <olivier@argent.xyz>
+ */
+contract RelayerModule is BaseModule {
+
+    bytes32 constant NAME = "RelayerModule";
+    uint256 constant internal BLOCKBOUND = 10000;
+
+    using SafeMath for uint256;
+
+    mapping (address => RelayerConfig) public relayer;
+
+    // The storage of the limit
+    ILimitStorage public limitStorage;
+    // The Token price storage
+    ITokenPriceStorage public tokenPriceStorage;
+
+    struct RelayerConfig {
+        uint256 nonce;
```

Comment:
The `nonce` can currently only store up to `uint128` max value and perhaps we should make that clear in the type here?

Reply:
Why do you say that? The nonce is made of 2 `uint128` so it needs to be a `uint256`.

Reply:
Sorry, thinking only of the first half of the nonce i.e. the block number but we're not saving that anywhere so ok.